### PR TITLE
fix: workflow syntax

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

### CODE REVIEW

File: .github/workflows/stable-snapshot-timestamp.yml [hunk 1]
```diff
@@ -39,11 +39,10 @@ jobs:
...
       contents: 'write'
-    steps:
-      - uses: sigstore/root-signing/.github/workflows/reuseable-snapshot-timestamp.yml@main
-        with:
-          snapshot_key: 'gcpkms://projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/snapshot'
-          timestamp_key: 'gcpkms://projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/timestamp'
-          repo: 'repository/'
-          provider: projects/237800849078/locations/global/workloadIdentityPools/root-signing-pool/providers/sigstore-root'
-          service_account: 'sigstore-root-signing@project-rekor.iam.gserviceaccount.com'
+    uses: sigstore/root-signing/.github/workflows/reuseable-snapshot-timestamp.yml@main
```
Comment (cpanato) [inline .github/workflows/stable-snapshot-timestamp.yml:41]: missing the steps and the array

```yaml
name: Stable Snapshot and Timestamp

permissions: read-all

# Execute this as a biweekly cron job and on changes to repository/
# when new published metadata is submitted.
on:
  # Enable cron for re-signing snapshot and timestamp every week
  schedule:
    - cron: '0 0 */7 * *' # every 7 days
  # When a new root is staged
  push:
    branches:
      - main
    paths:
      - 'repository/staged/root.json'
  workflow_dispatch:

jobs:
  run_snapshot_timestamp_publish:
    runs-on: ubuntu-20.04
    permissions:
      id-token: 'write'
      issues: 'write'
      pull-requests: 'write'
      contents: 'write'
    steps:
    - uses: sigstore/root-signing/.github/workflows/reuseable-snapshot-timestamp.yml@main
      with:
        snapshot_key: 'gcpkms://projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/snapshot'
        timestamp_key: 'gcpkms://projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/timestamp'
        repo: 'repository/'
        provider: projects/237800849078/locations/global/workloadIdentityPools/root-signing-pool/providers/sigstore-root'
        service_account: 'sigstore-root-signing@project-rekor.iam.gserviceaccount.com'
```

you can test the file here https://rhysd.github.io/actionlint/

Comment (asraa) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: interesting, It seemed like reusable workflows couldn't be invoked in steps... let me see.
https://docs.github.com/en/actions/using-workflows/reusing-workflows

Comment (cpanato) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: ahh, this file is meant to be reusable in another workflow? if yes we need to rewrite that

Comment (asraa) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: I think I see the issue: I can't use `runs-on` or `permissions`!

Comment (cpanato) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: and the `on`

Comment (asraa) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: Hmm this is the caller workflow: it should have the `on`

Comment (asraa) [reply .github/workflows/stable-snapshot-timestamp.yml:41]: So looking at this example https://docs.github.com/en/actions/using-workflows/reusing-workflows#example-caller-workflow

(1) reusbale workflows are not called in steps, they're called as jobs.job_id.uses
(2) I cannot provide a `runs-on`, but I can give explicit perms

**Review by cpanato — APPROVED** [review]

**Pull request merged**